pycyphal/transport/udp/_udp.py

```diff
@@ -247,6 +225,12 @@ def finalizer() -> None:
         return out
 
     def sample_statistics(self) -> UDPTransportStatistics:
+        # Update statistics for all keys in self._statistics.received_datagrams
+        for key in self._statistics.received_datagrams:
+            self._statistics.received_datagrams[key] = []  # Clear the old data
```

Comment:
This makes the old statistical counters disappear when a session is terminated, which mostly defeats the purpose of the statistics. Also, notice how the state kept in `self._statistics` is now effectively redundant since you destroy it right here and then re-construct it from scratch.

**State is bad.**

State should be avoided if at all possible, so in this case, a wise decision would be to remove `self._statistics` completely, but then we still have the first issue which is that the statistical counters are highly volatile and are not very useful because of that.

I recommend either solution:

- Just remove the statistics, for now, to fix it later.
- Create a new zero-initialized/empty entry in the `received_datagrams` dict when a new session is created using `self._statistics.received_datagrams.setdefault(...)` and pass a reference to it to the session constructor. Let the session update the counters while it exists. When the session is terminated, the corresponding entry in the `received_datagrams` dict is **not to be removed**.